core/src/test/scala/io/qbeast/core/transform/CDFNumericQuantilesTransformationTest.scala

```diff
@@ -24,9 +23,16 @@ class CDFNumericQuantilesTransformationTest extends AnyFlatSpec with Matchers {
     qt.transform(4) should be(1.0)
   }
 
-  it should "return correct transformation for insertion point in middle" in {
-    val qt = CDFNumericQuantilesTransformation(IndexedSeq(1.0, 2.0, 3.0), DoubleDataType)
-    qt.transform(1.5) should be(0.0)
+  it should "return correct transformation for insertion point in the bin" in {
+    val qt = CDFNumericQuantilesTransformation(IndexedSeq(1, 3, 5), orderedDataTypeTest)
+    // 2 is between 1 and 3, so it should be 0.25 (fraction = 2-1 / 3-1 = 0.5. -> 0 + fraction(0.5) / 2 = 0.25)
+    qt.transform(2) should be(0.25)
+  }
+
+  it should "return correct transformation for insertion point in the bin with repeated values" in {
```

Comment:
What do you expect to happen if a value falls with a bin that has the same min-max value?

Reply:
Sorry, I just realized it would be a `Found index`